specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json

```diff
@@ -0,0 +1,292 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-06-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/privateLinkResources": {
+      "get": {
+        "tags": [
+          "PrivateLinkResources"
+        ],
+        "description": "Gets the private link resources for SQL server.",
+        "operationId": "PrivateLinkResources_ListByServer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved private link resources.",
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkResourceListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 409 OperationCancelled - The operation has been cancelled by user.\n\n * 409 OperationInterrupted - The operation on the resource could not be completed because it was interrupted by another operation on the same resource.\n\n * 500 OperationTimedOut - The operation timed out and automatically rolled back. Please retry the operation."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets private link resources for SQL.": {
+            "$ref": "./examples/PrivateLinkResourcesList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/privateLinkResources/{groupName}": {
```

Comment:
how is a private link resource created?

Reply:
how is the groupName parameter in the URL different from the groupId parameter in the body?

Reply:
So this API doesn't map to any actual resource. This API is a utility to allow customers to be able to pass the proper "groupId" and "memberName" parameters to PUT PrivateEndpoint. Since groupId and memberName are different for each service, all implementing services have to implement this API. It is defined in section 6.4 in the [NRP Spec](https://microsoft.sharepoint.com/:w:/r/teams/WAG/AzureNetworking/Engineering%20Q%20%20Z/SDN-NRP/Design%20Specs/[NRP]%20Private%20Endpoint%20Design%203.docx?d=w64f10150563a47159476ffa24f9bd42e&csf=1&e=j5iuuM).

Since SQL has only one groupId and member, we just return static values.